x-pack/plugins/code/server/git_operations.ts

```diff
@@ -103,18 +103,16 @@ export class GitOperations {
     if (revision.toUpperCase() === 'HEAD') {
       return await repo.getHeadCommit();
     }
-    try {
-      return await repo.getBranchCommit(revision);
-    } catch (e) {
-      if (e.errno === Error.CODE.ENOTFOUND) {
-        return checkExists(
-          () => this.findCommit(repo, revision),
-          `revision or branch ${revision} not found in ${repo.path()}`
-        );
-      } else {
-        throw e;
-      }
+    // branches and tags
+    const refs = [`refs/remotes/origin/${revision}`, `refs/tags/${revision}`];
+    const commit = await this.findCommitByRefs(repo, refs);
```

Comment:
so I guess you could delete `getBranchCommit` method now?

Reply:
`getBranchCommit` is from NodeGit which doesn't work as we want, that's why we have this pr.